### PR TITLE
lease: add TTL() method

### PR DESCRIPTION
**Pull request by gyuho**
`Lease` is an exported type, so is TTL field. So anybody can modify `*lease.Lease`.

`lease` package internally has `refresh` method accessing this `TTL` field, so it can cause race conditions with external packages that use `*lease.Lease`.

Fix https://github.com/coreos/etcd/issues/6595.

/cc @xiang90 @heyitsanthony

### CODE REVIEW

**Comment by heyitsanthony** [comment]
Is the TTL only set when the lease is created except for that test case? I think the way it's being used now is safe.

It might be good to make `Lease.TTL`  private and export via a function `Lease.TTL()` to strictly enforce the constness across packages, though.

**Comment by gyuho** [comment]
@heyitsanthony Agree. I will make TTL field private. Currently we only read TTL field and modify just for testing purposes. Will drop `SetTTL` method.

File: lease/lessor_test.go [hunk 1]
```diff
@@ -140,13 +140,13 @@ func TestLessorRenew(t *testing.T) {
 	}
 
 	// manually change the ttl field
-	l.TTL = 10
+	l.SetTTL(10)
```
Comment (heyitsanthony) [inline lease/lessor_test.go]: Would this fix the race?
```go
le.mu.Lock()
l.TTL = 10
le.mu.Unlock()
```

Comment (gyuho) [inline lease/lessor_test.go]: Right, just fixed. `refresh` method calls are already protected with mutexes.

Just kept `TTL()` method for external packages.

PTAL. Thanks.

**Comment by xiang90** [comment]
LGTM. defer to @heyitsanthony

File: lease/lessor.go [hunk 2]
```diff
@@ -484,23 +484,25 @@ func (le *lessor) initAndRecover() {
 }
 
 type Lease struct {
-	ID  LeaseID
-	TTL int64 // time to live in seconds
+	ID LeaseID
+
+	mu  sync.Mutex // to protect TTL
```
Comment (heyitsanthony) [inline lease/lessor.go]: This isn't necessary. If TTL is only set at creation time, there's no data race when fetching it thereafter.

Comment (gyuho) [inline lease/lessor.go]: Right. Just fixed. PTAL.

**Comment by xiang90** [comment]
@gyuho Then why do we need to change TTL to ttl and add a getter func?

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
@xiang90 To prevent external package from modifying `*lease.Lease` struct. This change would make `TTL` field read-only for external packages. We return by pointer (`Grant`, `Lookup`). So external projects can access TTL exported field with races, while `lease` package reads this field with proper locking.

**Comment by xiang90** [comment]
OK. Sounds good.

**Pull request merged**